backend/src/api/websocket-handler.ts

```diff
@@ -195,21 +201,46 @@ class WebsocketHandler {
           }
 
           if (parsedMessage && parsedMessage['track-address']) {
-            if (/^([a-km-zA-HJ-NP-Z1-9]{26,35}|[a-km-zA-HJ-NP-Z1-9]{80}|[a-z]{2,5}1[ac-hj-np-z02-9]{8,100}|[A-Z]{2,5}1[AC-HJ-NP-Z02-9]{8,100}|04[a-fA-F0-9]{128}|(02|03)[a-fA-F0-9]{64})$/
-              .test(parsedMessage['track-address'])) {
-              let matchedAddress = parsedMessage['track-address'];
-              if (/^[A-Z]{2,5}1[AC-HJ-NP-Z02-9]{8,100}$/.test(parsedMessage['track-address'])) {
-                matchedAddress = matchedAddress.toLowerCase();
+            const validAddress = this.testAddress(parsedMessage['track-address']);
+            if (validAddress) {
+              client['track-address'] = validAddress;
+            } else {
+              client['track-address'] = null;
+            }
+          }
+
+          if (parsedMessage && parsedMessage['track-addresses'] && Array.isArray(parsedMessage['track-addresses'])) {
+            const addressMap: { [address: string]: string } = {};
+            for (const address of parsedMessage['track-addresses']) {
+              const validAddress = this.testAddress(address);
+              if (validAddress) {
+                addressMap[address] = validAddress;
               }
-              if (/^04[a-fA-F0-9]{128}$/.test(parsedMessage['track-address'])) {
-                client['track-address'] = '41' + matchedAddress + 'ac';
-              } else if (/^(02|03)[a-fA-F0-9]{64}$/.test(parsedMessage['track-address'])) {
-                client['track-address'] = '21' + matchedAddress + 'ac';
-              } else {
-                client['track-address'] = matchedAddress;
+            }
+            if (Object.keys(addressMap).length > config.MEMPOOL.MAX_TRACKED_ADDRESSES) {
+              response['track-addresses-error'] = `too many addresses requested, this connection supports tracking a maximum of ${config.MEMPOOL.MAX_TRACKED_ADDRESSES} addresses`;
```

Comment:
This is not sent back as valid JSON like other data.

`Unexpected token 'o', ..."-error": too many ad"... is not valid JSON`

Reply:
Fixed